From c63d7e7a3e4e6aa5016715d599a9b6ebc7481fba Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 17 Oct 2022 16:40:29 +0200
Subject: [PATCH 06/10] FROM ARCHLINUX: upate xclient script
X-Developer-Signature: v=1; a=openpgp-sha256; l=1506; i=xtexchooser@duck.com;
 h=from:subject; bh=P5my4cfLg6bWALX22OnVo9qf4RtNBBBACVU67U9Dp9M=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDGkvVLaGBH3fv01kmr64ocs2mRCLYE6++ECmWZG3Dq83L
 XsXWivWUcrCIMbFICumyFJk2ODNqpPOL7qsXBZmDisTyBAGLk4BmMi0cwz/Y/vlYr8c5LDLN2g1
 itr0+d5WlbhMf7MVfoVJRm5hQYzfGBlua+eyfjh/ueLmQxXeueIHjvWu7c97nRQ14w2vNRNb7V1
 eAA==
X-Developer-Key: i=xtexchooser@duck.com; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

---
 src/VBox/Additions/x11/Installer/98vboxadd-xclient | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/VBox/Additions/x11/Installer/98vboxadd-xclient b/src/VBox/Additions/x11/Installer/98vboxadd-xclient
index 7a45de98a7ec..dcba5fd26436 100755
--- a/src/VBox/Additions/x11/Installer/98vboxadd-xclient
+++ b/src/VBox/Additions/x11/Installer/98vboxadd-xclient
@@ -31,12 +31,9 @@ for i in $HOME/.vboxclient-*.pid; do
     test -w $i || rm -f $i
 done
 
-if ! test -c /dev/vboxguest 2>/dev/null; then
-   # Do not start if the kernel module is not present.
-   # Execute notify-send in the back-ground to avoid racing with sddm,
-   # as notify-send may wait for sddm to start while it waits for us to exit.
-   notify-send "VBoxClient: the VirtualBox kernel service is not running.  Exiting." &
-elif test -z "${SSH_CONNECTION}"; then
+# Do not start if the kernel module is not present; or if this script is
+# triggered by a connection over SSH.
+if [ -c /dev/vboxguest ] && [ -z "${SSH_CONNECTION}" ]; then
    # This script can also be triggered by a connection over SSH, which is not
    # what we had in mind, so we do not start VBoxClient in that case.  We do
    # not use "exit" here as this script is "source"d, not executed.
@@ -52,5 +49,4 @@ elif test -z "${SSH_CONNECTION}"; then
     fi
 
     /usr/bin/VBoxClient --checkhostversion
-    /usr/bin/VBoxClient --vmsvga-session # In case VMSVGA emulation is enabled
 fi
-- 
2.46.0

